### PR TITLE
[2.20.x] DDF-5649 G-6176 Add interface to filter warnings (#5843)

### PR DESCRIPTION
**Pull request by jlcsmith**
Abbreviated Review
Backport of https://github.com/codice/ddf/pull/5843
Note the second commit was an addition to the original PR to route processing details all the way back to the UI.

#### Who is reviewing it? 
@kentmorrissey 
@derekwilhelm 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining 
@pklinef  

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java [hunk 1]
```diff
@@ -551,7 +551,13 @@ public CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
                 .filter(Objects::nonNull)
                 .map(QueryResponse::getProperties)
                 .findFirst()
-                .orElse(Collections.emptyMap()));
+                .orElse(Collections.emptyMap()),
+            responses
+                .stream()
+                .filter(Objects::nonNull)
+                .map(QueryResponse::getProcessingDetails)
+                .findFirst()
```
Comment (rzwiefel) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: Not sure the context, but just making sure we are fine finding only the first and not using the rest? We don't need to collapse/reduce them into a single set?

Comment (jlcsmith) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: good catch - fixed

**Pull request merged**